extensions/mongodb-client/deployment/src/main/java/io/quarkus/mongodb/deployment/MongoClientProcessor.java

```diff
@@ -371,4 +368,20 @@ HealthBuildItem addHealthCheck(MongoClientBuildTimeConfig buildTimeConfig) {
         return new HealthBuildItem("io.quarkus.mongodb.health.MongoHealthCheck",
                 buildTimeConfig.healthEnabled);
     }
-}
+
+    @BuildStep
+    void registerServiceBinding(Capabilities capabilities, BuildProducer<ServiceProviderBuildItem> buildProducer) {
+        if (capabilities.isPresent(Capability.KUBERNETES_SERVICE_BINDING)) {
+            buildProducer.produce(
+                    new ServiceProviderBuildItem(SERVICE_BINDING_INTERFACE_NAME,
+                            MongoServiceBindingConverter.class.getName()));
+        }
+    }
+
+    // TODO(SG): do we need this instead of the other registerServiceBinding?
+    @BuildStep
+    RunTimeConfigurationSourceValueBuildItem registerServiceBinding(KubernetesServiceBindingRecorder recorder,
+            KubernetesServiceBindingConfig config) {
+        return new RunTimeConfigurationSourceValueBuildItem(recorder.configSources(config));
+    }
```

Comment:
This should not be needed as the `quarkus-kubernetes-service-binding` creates them. From your POV, all that is needed is to register `io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConverter`.
That said, I think you are missing a `META-INF/services/io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConverter` like https://github.com/quarkusio/quarkus/blob/2.1.0.Final/extensions/kafka-client/runtime/src/main/resources/META-INF/services/io.quarkus.kubernetes.service.binding.runtime.ServiceBindingConverter

Reply:
I thought I added it in (at the very end of this page)

Reply:
Good point, it was too small in the UI so I missed it. In any case, the rest of the comment stands